learners/style.md

```diff
@@ -0,0 +1,116 @@
+---
+title: 'Style Guide'
+---
+
+A collection of guidelines to follow when writing
+source (R) Markdown files for use with {sandpaper}.
+The Carpentries also provides a [general Style Guide](https://docs.carpentries.org/topic_folders/communications/resources/style-guide.html)
+for written content e.g. blog posts, website and lesson content, etc.
+
+- [Fenced Divs](#fenced-divs)
+  - [Fence Length](#fence-length)
+  - [Whitespace](#whitespace)
+- [Figures](#figures)
+- [Line Length](#line-length)
+
+## Fenced Divs
+
+### Fence Length
+
+For increased legibility, we recommend an opening fence length of at least 20 colons
+(approximately one third of the screen), 
+with a single space between the last colon and the class keyword.
+The length of closing fences should match the length of the opening fence plus keyword,
+for example:
+
+```markdown
+:::::::::::::::::::: callout
+
+### An example callout
+
+This callout was opened with a fence of twenty colons.
+The end of the closing fence aligns with
+the last character in the 'callout' keyword.
+
+::::::::::::::::::::::::::::
+```
+
+Where fenced divs are nested,
+e.g. attaching a solution block to a challenge,
+we recommend making the inner fences at least ten colons longer
+than the outer fences.
+E.g.
+
+```markdown
+:::::::::::::::::::: challenge
+
+### An example challenge
+
+This challenge was opened with a fence of 20 colons.
+
+:::::::::::::::::::::::::::::::::: solution
+
+The nested solution block was opened with 35 colons,
+to make it stand out from the challenge.
+
+:::::::::::::::::::::::::::::::::::::::::::
```

Comment:
I'm wondering about this suggestion. I do not disagree with it per se, but I used the _opposite_ rule:

outer blocks use more colons than inner blocks. For example, in [the conversion process from {pegboard}](https://github.com/carpentries/pegboard/blob/3b06f4559892e31ff0a6bae1d838cd1ba6fd94fb/R/div.R#L14), we converted solutions to use 25 colons and everything else to use 50. 

I apologise for the lateness of this information, but past Zhian wrote something of a style guide many moons ago: https://carpentries.github.io/sandpaper/articles/landscape.html#fenced-div-tags

Though feel free to take this with a grain of salt. I trust your and @bencomp's judgements

Reply:
I had noticed this pattern in the converted lessons. And, on the one hand, there is value in trying to remain consistent with the style that already exists in lessons.

On the other hand, I believe most people will be typing the string of colons when they add a new fence, and it takes less time to add 20 colons than it does to add 50.

On reflection, I think consistency wins out here, so I will change the recommendation.

BTW the recommendation to use full-width fences for Instructor Notes is a good one - I wish I'd read that before the Drive last month!

Reply:
> BTW the recommendation to use full-width fences for Instructor Notes is a good one - I wish I'd read that before the Drive last month!

🙈